tests/e2e/utils/service_utils.go

```diff
@@ -85,6 +86,9 @@ func WaitServiceExposureAndValidateConnectivity(cs clientset.Interface, namespac
 	if err != nil {
 		return "", err
 	}
+	if service == nil {
+		return "", errors.New("the service is nil")
+	}
```

Comment:
A little curious, when will `svc == nil` and `err == nil` be returned? Should it return https://github.com/nilo19/cloud-provider-azure/blob/d5e66b600ba6464c5cb490f9824bb4919b0fa75e/vendor/k8s.io/apimachinery/pkg/api/errors/errors.go#L527 ?

Reply:
I don't understand either. But it's panicked here.

Reply:
Here might be an issue. Consider this:
1. poll fn runs and goes to [L135](https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/tests/e2e/utils/service_utils.go#L135) (which `service != nil` and `err == nil`)
2. and poll fn reaches timeout, then [L144](https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/tests/e2e/utils/service_utils.go#L140) would return`nil, nil` (But it should return timeout?

Do I understand this right?

Reply:
it will return a timeout error then

Reply:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cloud-provider-azure/1494/pull-cloud-provider-azure-e2e-ccm-capz/1518841182073917440

Reply:
> it will return a timeout error then

L144 the `err` variable seems not to be assigned to `ErrTimeout`, the condition L135 checks the err returned by poll fn instead.

And the log shows that it keeps going into the [branch](https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/tests/e2e/utils/service_utils.go#L133-L136)

Reply:
right, thanks for catching